public/src/unityIAP/java/com/revenuecat/purchases/models/StoreProduct.kt

```diff
@@ -136,7 +136,7 @@ data class StoreProduct(
      *
      * For Amazon, the original Product can be obtained using `StoreProduct.amazonProduct`
      */
-    val originalJson: JSONObject
```

Comment:
one thought -- this class is a public interface, and really only serves as a data class. i wonder if it'd be possible to keep the interface as much the same as possible.

for instance, perhaps we can get the freeTrialPeriod from the pricingphases. then the only things that need to be split between BC5-backwards-compat and non-backwards-compat is the `storeProductConversions`

but we will have to play around with that, it might not make sense

Reply:
I like the idea a lot. I understand both productDetails and offerToken are opaque pieces of data we just need to pass on to the purchasing flow when we call BC5 for buying. We could solve this by just having this as a BC5 specific subclass, right? We can internally downcast whenever without having to expose this information publicly.

The only thing that concerns me is that BC5 with pricingPhases has a richer interface, right?
So while we can map **some** pricingPhases into free, intro, and price model, it might not apply to all of them.
Similar to what we had discussed here: https://revenuecat.slack.com/archives/C02CD1TLS76/p1663934586310319 - I think it would make sense to have a new interface for pricingPhases, So I think we should add them here.

Should we maybe even consider going the other way? Add a new pricingPhases and deprecate the old 3-phase model?